apps/libsignpost/signpost_api.h

```diff
@@ -261,23 +265,39 @@ int signpost_processing_reply(uint8_t src_addr, uint8_t message_type, uint8_t* r
 /* ENERGY API                                                             */
 /**************************************************************************/
 
+//message types
 enum energy_message_type {
     EnergyQueryMessage = 0,
     EnergyLevelWarning24hMessage = 1,
     EnergyLevelCritical24hMessage = 2,
     EnergyCurrentWarning60sMessage = 3,
+    EnergyReportMessage = 4,
+    EnergyDutyCycleMessage = 5,
 };
 
+//information sent to a module from the controller
 typedef struct __attribute__((packed)) energy_information {
-    uint32_t    energy_limit_24h_mJ;
-    uint32_t    energy_used_24h_mJ;
-    uint16_t    current_limit_60s_mA;
-    uint16_t    current_average_60s_mA;
+    uint32_t    energy_limit_mWh;
+    uint16_t    average_power_mW;
```

Comment:
Ok that's reasonable. What is the suggested lifetime that a module should use? I might think, "well, we'll probably get more energy sometime over the week, so I'll set my lifetime to 7 days and see how often I can sample." I can also see, "well, my study is supposed to last 6 months, and I have no idea when energy will come in, so I'll set my lifetime to 6 months and sample at that rate." (which would likely end up with the module taking progressively more samples as the 6 month period gets closer to over). Obviously those are going to give very different results.

So while this is simple (only two values) it may not be all that simple from a module's point of view since the controller is doing no work to estimate when new energy might be available.